scripts/kruise-kontrol-wrapper.lua

```diff
@@ -1,244 +1,115 @@
+      local p = game.players[pindex]
+      local c = p and p.valid and p.character
+
+      -- But, actually, if there's no player entity and that player entity
+      -- doesn't have a character, then we're done.  Can't do KK for that.
+      if not c.valid then return end
+
+      local hand = c.cursor_stack
+      if hand and hand.valid_for_read and (hand.name == "blueprint" or hand.name == "blueprint-book") then
+         return { "access.kk-blueprints-not-allowed" }
       end
 
-      if entity.type == "tile-ghost" then
-         status = "building ghosts"
-         return status
-      end
+      -- Okay. Finally we're good.  Let's kick this off.
+      remote.call(interface_name, "start_job", pindex, { x = math.floor(kk_pos.x), y = math.floor(kk_pos.y) }, target)
+      local desc = remote.call(interface_name, "get_description", pindex)
+      if not desc then return { "access.kk-not-started" } end
 
-      if entity.to_be_deconstructed() then
-         status = "deconstructing"
-         return status
-      end
+      return { "access.kk-start", desc }
+   end, { "access.kk-not-available" })
 
-      if entity.get_health_ratio() and entity.get_health_ratio() < 1 then
-         status = "repairing"
-         players[pindex].kk_radius = 50
-         return status
-      end
+   printout(announcing, pindex)
+end
 
-      if entity.to_be_upgraded() then
-         status = "upgrading"
-         return status
+--FA actions to take when KK cancel input is pressed
+function mod.cancel_kk(pindex)
+   call_with_interface(function()
+      if not remote.call(interface_name, "is_active", pindex) then
+         -- If there was no interface then KK isn't installed; if the player
+         -- isn't active already then the enter key is doing driving.  Nothing
+         -- to say.
+         return
       end
 
-      local fuel_inventory = entity.get_fuel_inventory()
-      if fuel_inventory and fuel_inventory.is_empty() then
-         status = "refueling"
-         players[pindex].kk_radius = 50
-         return status
-      end
-   end
-
-   if entity.to_be_deconstructed() and (force.name == "neutral") then
-      status = "deconstructing"
-      return status
-   end
-
-   if entity.type == "resource" then
-      status = "mining resources"
-      return status
-   end
-
-   if entity.type == "tree" then
-      status = "mining resources"
-      return status
-   end
-
-   if entity.type == "simple-entity" and force.name == "neutral" then
-      status = "mining resources"
-      return status
-   end
-
-   if p.force.get_cease_fire(entity.force) == false then
-      status = "attacking"
-      players[pindex].kk_radius = 64
-      return status
-   end
-
-   if entity.train ~= nil or entity.type == "car" then
-      status = "following"
-      return status
-   end
-
-   --Unknown case:
-   return "walking"
-end
+      -- We screwed around with the running modifier. Put it back based on
+      -- cursor mode.
+      fix_walk(pindex)
```

Comment:
Info: The thing about messing with walking is that telestep mode goes bonkers in multiplayer, so in this walking mode we intentionally force the player walking speed to become zero. 
The new version of this file handles this matter correctly.